### ISSUE TITLE
Suggestion: Polyrhythm PPQN modes  allow BPM change

### ISSUE THREAD

**Issue by rasavell** [jeremymuller/Sha-Bang-Modules#67]
Nice module! Works great! Would be nice to manually vary bpm when external triggering. But maybe there's some programming complication that I don't see. ???  Thanks for the great modules!

**Comment by jeremymuller** [comment]
What do you mean manually vary the bpm while external triggers? Using the external clock will override the internal clock, and you can vary the bpm however you want from the ext clock.

**Comment by rasavell** [comment]
Well, that doesn’t seem to be how the module is working for me. It doesn’t track the bpm of the external clock. It seems to behave as stated in your doc:

If the mode is set to any of the PPQN modes, the clock will turn on automatically when it receives a pulse. It will also turn off automatically after it times out from not receiving any more pulses.
So the unit toggles on and off when it gets external clock pulses in the PPQN mode and while it’s on it generates pulses at the . And the bpm can’t be varied in this mode once you hook up the external clock. Seems like it would be a useful mode if you could vary the bpm while the external clock is hooked up. 

Like I said, that’s how it seems to be working for me.

Rob

**Comment by Petervos2018** [comment]
> Seems like it would be a useful mode if you could vary the bpm while the external clock is hooked up.

The only way to change the bpm when the external clock is hooked up is, to change the external clock, the external clock determines the BPM.

**Comment by rasavell** [comment]
Not what your manual says and not how the unit works. But that’s cool. Later.

**Comment by jeremymuller** [comment]
@rasavell, ok it's my manual, not @Petervos2018. And he was helping explain to you what you said the module doesn't do. The module will literally change the BPM based on the external clock (what @Petervos2018 was talking about and what you've been asking about). And the manual also explicitly says this. 

See for yourself:
[external clock literally changing the BPM of PolyrhythmClock](https://youtu.be/CBgvvASdz0g)

**Comment by rasavell** [comment]
Okay,
The module I’m using (Rack2- library up to date) has a bug and doesn’t track the External Clock for 2PPQN.  (see below. Setting is 2PPQN). It does track for the other modes. That is the source of the confusion. 

I really do like your modules. And like I said before. Thanks for all the hard work and making them available. 
Best, Rob

**Comment by jeremymuller** [comment]
Yes that's a known issue. I'm working on that clock as well as others that use the same thing. I'll include that fix in the update.
